packages/core/src/compiler.ts

```diff
@@ -65,17 +65,21 @@ export default class Compiler {
    * @param src - The source for which the JS code should be generated
    * @param plural - The default locale
    * @param plurals - A map of pluralization keys for all available locales
+   * @param isDescendentCall - Indicates whether or not this is the top level of the recursive call
    */
   compile(
     src: string | StringStructure,
     plural: PluralObject,
-    plurals?: { [key: string]: PluralObject }
+    plurals?: { [key: string]: PluralObject },
+    isDescendentCall?: boolean
   ) {
     if (typeof src === 'object') {
       const result: StringStructure = {};
       for (const key of Object.keys(src)) {
-        const pl = (plurals && plurals[key]) || plural;
-        result[key] = this.compile(src[key], pl, plurals);
+        const pl = isDescendentCall
+          ? plural
+          : (plurals && plurals[key]) || plural;
+        result[key] = this.compile(src[key], pl, plurals, true);
```

Comment:
```suggestion
        const pk = plurals && plurals[key];
        result[key] = pk
          ? this.compile(src[key], pk)
          : this.compile(src[key], pl, plurals);
```

Wouldn't that have the same effect, with no need for an additional argument?

Reply:
Hi, sorry for the delayed response! 

Yes, I believe this would work. It's slightly different in that it supports locale keys at any depth... I didn't realize at first, but your feedback on the other PR leads me to believe that this is a documented feature?

In other words, should the following specification pass?

```
    it('should produce the correct output given nested sub keys that look like plurals', () => {
      const mf = new MessageFormat('*');

      const messagePacks = {
        someGrouping: {
          es: {
            en: 'Estimation Date: {date, date, ::EEEMMMd}',
            fi: 'Financial Incentive Date: {date, date, ::EEEMMMd}'
          }
        },
      };

      const result = compileModule(mf, messagePacks);

      expect(result).toMatch(`new Intl.DateTimeFormat("es", opt);`);

      expect(result).not.toMatch(`new Intl.DateTimeFormat("fi", opt);`);
      expect(result).not.toMatch(`new Intl.DateTimeFormat("en", opt);`);
    });
```

If yes then we should take your suggestions.

Reply:
@dpchamps Sorry, completely dropped the ball on this. Yes, locale keys may be set at any depth. For instance, you could have a structure like this, and it should work:
```js
{ foo: { bar: { en: 'Bar in English', fi: 'Baari suomeksi' } } }
```

Reply:
@eemeli Now I've gotta apologize! To be quite honest, I don't remember the context for this fix, so I'll have to set aside some time for focus. 

Are you still interested in landing this patch? If so, I can take a look sometime this week.

Reply:
I'd still be interested in landing something like this, and would be happy to review an update. No hurry on my part though.